[Settings] Various UX tweaks

niels9001 opened the pull request:
## Summary of the Pull Request
- Re-organized the General page (#2804 #2688)
- Applied consistent add button styling (for KBM and ImageResizer( #2806 #2710)
- Increased the default step value for the shortcut transition time (#2808)
- Changed the 'About feature' text to 'About [MODULE NAME]' (#2805)
- Added a direct link to this GitHub repo on the General page.
- Minor localization fixes.


General page:
![image](https://user-images.githubusercontent.com/9866362/81483649-0e146000-9240-11ea-86be-1c6e1d56a856.png)

New Add button style:
![Addbutton](https://user-images.githubusercontent.com/9866362/81483653-1371aa80-9240-11ea-8ccf-bd88e9baa621.gif)


## References
#2804 #2688 #2806 #2710 #2808 #2805


## PR Checklist
* [X] Applies to #xxx
* [X] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA
* [] Tests added/passed
* [] Requires documentation to be updated
* [] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

htcfreek commented:
@niels9001 Thank you for making this changes.

Here are my thought about the changes.

**Image resizer settings**
I don't like the add button style. Looks really strange under the table and is not in line with the other buttons.
But that is my opinion.

**General settings page**
- The automatic startup setting looks like a sub item of admin mode setting. We shoul not have the "running ad user in bold.)
- What about writing the hint "Restart as admin to change this" in a seperate line betwen caption and control, without brakets and in red/yellow?
- We should add a text hint that auto download only works when running in admin mode and not using a metered connection.
- What about having a seperate "Release Notes" link under the "check update" button?
- What about split side pannel in three sections:

>**About PowerToys project**
PowerToys is open source software.
[Learn more](https://github.com/microsoft/PowerToys/blob/master/README.md#contributing)
[Github project page](https://github.com/microsoft/PowerToys)
>
>**Give feedback**
[Request a feature]()
[Report a Bug)
>
>**Privacy**
>We collect Telemetry data. But we respect your privacy.
[Privacy statement](https://github.com/microsoft/PowerToys/blob/master/README.md#privacy-statement)

htcfreek commented:
@crutkas I have understood that we don't allow disabling the telemetry at the moment.
But should we implement the settings and hide them for now, that we don't forget the mock up ideas or should we write a comment for that within the related issue?

crutkas commented:
My focus right now is on 0.18. I don’t want it there until we have the hooks also in place to properly run it. This is a v1 item with its own tracking item.

mdtauk commented:
The width of the hover on the Add Button, should align to the grid above it.

htcfreek commented:
@niels9001 Is the link "Learn more about admin mode" hidden, if the settings run in admin mode? I think it should be shown every time because we have the "Always start as admin" setting.

laviusmotileng-ms commented:
> @niels9001 Is the link "Learn more about admin mode" hidden, if the settings run in admin mode? I think it should be shown every time because we have the "Always start as admin" setting.

looking at the code, I think it will always show.

niels9001 commented:
@mdtauk What was the name of the brush again that does the red (for light) and yellow (for dark) warning text?

mdtauk commented:
**SystemControlErrorTextForegroundBrush**

niels9001 commented:
@mdtauk Good catch, I have increased the width so it's inline with the listview.
![Addbuttongif](https://user-images.githubusercontent.com/9866362/81483604-b4139a80-923f-11ea-9f2a-21ef890c3bb9.gif)

@htcfreek I have made some changes based on your feedback. The warning label should be collapsed when running as admin, but let's track that in a separate issue.

![image](https://user-images.githubusercontent.com/9866362/81483644-fc32bd00-923f-11ea-8fea-e8c2615c14fb.png)

htcfreek commented:
> @htcfreek I have made some changes based on your feedback. The warning label should be collapsed when running as admin, but let's track that in a separate issue.
> 
> ![image](https://user-images.githubusercontent.com/9866362/81483644-fc32bd00-923f-11ea-8fea-e8c2615c14fb.png)
> 
> 

@niels9001 
I like the new "Apperance & behavior" section. 👍

Please add a point behind the hint "Restart as admin". It's a sentence.

But what is whith the changes on the Updates section?
- Especially the issue #2354 which is fixed for the old settings.
- I think for the user it is important to know the infos in the auto download hint. So they understand why the updates won't be downloaded in the mentiond szenarios.

Du you have a reason why not restructuring the side pannel?

niels9001 commented:
> > @htcfreek I have made some changes based on your feedback. The warning label should be collapsed when running as admin, but let's track that in a separate issue.
> > ![image](https://user-images.githubusercontent.com/9866362/81483644-fc32bd00-923f-11ea-8fea-e8c2615c14fb.png)
> 
> @niels9001
> I like the new "Apperance & behavior" section. 👍
> 
> Please add a point behind the hint "Restart as admin". It's a sentence.
> 
> But what is whith the changes on the Updates section?
> 
> * Especially the issue #2354 which is fixed for the old settings.
> * I think for the user it is important to know the infos in the auto download hint. So they understand why the updates won't be downloaded in the mentiond szenarios.
> 
> Du you have a reason why not restructuring the side panel?

W10 Settings is not using periods at the end of an error label?
![image](https://user-images.githubusercontent.com/9866362/81484192-08207e00-9244-11ea-845c-e9515ec8f5b1.png)

@laviusmotileng-ms Do we have the logic in place that disable the 'auto update' toggleswitch? If so, we can add the same warning label.

Side panel: let's get feedback of @crutkas on this first.

htcfreek commented:
@niels9001 
We use different words for "administrator": admin, administrator
We should only use one.

> W10 Settings is not using periods at the end of an error label?
Oops. Never noted bevor. 🙈

niels9001 commented:
> @niels9001
> We use different words for "administrator": admin, administrator
> We should only use one.

Fixed in latest commit

htcfreek commented:
> > @niels9001
> > We use different words for "administrator": admin, administrator
> > We should only use one.
> 
> Fixed in latest commit
> 
> 

You have add a spelling mistake in line 489: `wncoder`

mdtauk commented on src/core/Microsoft.PowerToys.Settings.UI/Strings/en-us/Resources.resw:
```diff
@@ -486,22 +486,22 @@
   </data>
   <data name="ImageResizer_FallbackEncoder_BMP.Content" xml:space="preserve">
-    <value>BMP Encoder</value>
+    <value>BMP wncoder</value>
```
Typo here **wncoder**

niels9001 replied:
Thanks! Fixed in latest commit

htcfreek commented:
> The warning label should be collapsed when running as admin, but let's track that in a separate issue.

I have opened the issue #2829.

htcfreek commented:
Can we remove the 'v' in `Version v0.17.0`? 'v' has the same meaning like the word 'Version'. So the 'v' is one 'version' to much.

crutkas commented:
Let’s move discussions like this off the PR. 

Work was agreed upon, if DCRs are being asked for, let’s do that back on main topic. Else stuff willl never close out.

crutkas commented:
Just noticed text lingo here for running as admin, it is awkward. I think we should open a new PR and move adjust for 0.19

htcfreek commented:
> Just noticed text lingo here for running as admin, it is awkward. I think we should open a new PR and move adjust for 0.19

Do you mean the word "admin"? It's changed to Administrator in one of this PR's commits to use the same word in all texts.

niels9001 commented:
@crutkas @htcfreek A native English speaker should check the updated strings to make sure they are correct. I went for 'administrator' dur to the fact that that is what the Windows Shell (e.g. 'Run as administrator') and Settings are using.

htcfreek commented:
> @crutkas @htcfreek A native English speaker should check the updated strings to make sure they are correct. I went for 'administrator' dur to the fact that that is what the Windows Shell (e.g. 'Run as administrator') and Settings are using.

I am also not a native speaker. @crutkas  should do that.

htcfreek commented:
> Can we remove the 'v' in `Version v0.17.0`? 'v' has the same meaning like the word 'Version'. So the 'v' is one 'version' to much.
> 
> We also should also remove the underlining of the version number to be in line with the other links.

@niels9001 Can you please change this.

laviusmotileng-ms commented on src/core/Microsoft.PowerToys.Settings.UI/ViewModels/GeneralViewModel.cs:
```diff
@@ -126,26 +126,6 @@ public bool Packaged
             }
         }
 
-        public string AlwaysRunAsAdminText
```
remove line 183 too

laviusmotileng-ms replied:
I think tests also need to be updated for this.

htcfreek commented:
> Text doesnt change even when admin mode is enabled: 
> ![image](https://user-images.githubusercontent.com/58791731/81509340-be28ae00-92be-11ea-80bd-04d772b2a7b0.png)
> 
> 

See #2829

niels9001 commented:
@laviusmotileng-ms The idea is to set the Visibility to Collapsed when the app is running as admin.

Can I use the Elevated bool for that?

niels9001 commented:
@laviusmotileng-ms 

I have updated the code to fix the request. I have put in a converter now to hide the warning texts when running as admin.

If I understood correctly, the automatic downloads requires the same permissions, correct? I have added the same warning text there, and will be collapsed when running as admin.

Not running as admin
![image](https://user-images.githubusercontent.com/9866362/81510600-34ef9680-9313-11ea-8ef6-31aede28d49f.png)

Running as admin
![image](https://user-images.githubusercontent.com/9866362/81510629-797b3200-9313-11ea-836d-ee2107f23b68.png)

htcfreek commented:
@niels9001 
I am not shure. But it seems that the spacing in the Admin section is incorrect when running as Admin.

Can you please remove the 'v' in the version nunber. @laviusmotileng-ms has agreed to this change as you can see above.

Personally I'm not like the word use in the hint. I think 'change' is the better word.

laviusmotileng-ms commented:
> @laviusmotileng-ms The idea is to set the Visibility to Collapsed when the app is running as admin.

Okay I can push a quick PR against that.

riverar commented:
Do not use "Administrator". The user doesn't need to be an administrator, they just need an elevated token. I'd remove the hint completely and just use a standard UAC shield next to a toggle with the label `Always run elevated`. If the user has the appropriate privileges, the action will simply execute. If they don't, credential UI will appear. That should clean up that area nicely.

enricogior commented:
@riverar 
I agree, the "administrator" term is misleading and technically wrong.
Regarding allowing the feature for non-admin user we had a few issues, of course we should fix them and allow a user to run it elevated.
We don't have the time to do it in 0.18, but we should consider it for 0.19.

niels9001 commented:
Apologies for misunderstanding, this was based on existing strings that mentioned 'run as administrator'. If someone can provide the correct labels for all items in the screenshot above that'd be great! Just so I can put them in place!

Also: do we need to show the 'Restart as ...' in disabled state if the user has done so? We could hide that one as well right?

@riverar Adding the UAC icon is a good idea, since it's well recognizable across the OS.

htcfreek commented:
> Do not use "Administrator". The user doesn't need to be in the default Administrator group, which could cause confusion. They just need an elevated token. I'd remove the hint completely and just use a standard UAC shield next to a toggle with the label `Always run elevated`. That should clean up that area nicely.

The use of elevated shield is against the design guidlines.

htcfreek commented:
> Apologies for misunderstanding, this was based on existing strings that mentioned 'run as administrator'. If someone can provide the correct labels for all items in the screenshot above that'd be great! Just so I can put them in place!
> 
> Also: do we need to show the 'Restart as ...' in disabled state if the user has done so? We could hide that one as well right?
> 
> @riverar Adding the UAC icon is a good idea, since it's well recognizable across the OS.
> 

Hiding controls is against the design guidlines too.

htcfreek commented:
Windows itself uses the term "Run as Administrator".

I am not shure if normal users understand the term "run elevated".

enricogior commented:
@niels9001
I should have explained it better, we are not going to change anything for 0.18, and you correctly adopted the same language/functionality as the "old" Settings, but this needs to be reviewed after 0.18.

htcfreek commented:
> @niels9001
> I should have explained it better, we are not going to change anything for 0.18, and you correctly adopted the same language/functionality as the "old" Settings, but this needs to be reviewed after 0.18.
> 
> 

@enricogior I think we should open a new issue for that, which includes the ideas in this PR and mark it with "Project 0.19".

htcfreek commented:
> > Can we remove the 'v' in `Version v0.17.0`? 'v' has the same meaning like the word 'Version'. So the 'v' is one 'version' to much.
> > 
> > We also should also remove the underlining of the version number to be in line with the other links.
> 
> @niels9001 Can you please change this.

@niels9001 
I have seen that the 'v' is included in the version number string. So we can leave it as it is.

But where comes the underlining on the link? I don't find the formatting attribute in the source code who do this underlining.

riverar commented:
> > @riverar Adding the UAC icon is a good idea, since it's well recognizable across the OS.
> > 
> 
> The use of elevated shield is against the design guidlines. If controls were disabled, a info text should be existing.

Which design guidelines?

htcfreek commented:
> > > @riverar Adding the UAC icon is a good idea, since it's well recognizable across the OS.
> > > 
> > 
> > The use of elevated shield is against the design guidlines. If controls were disabled, a info text should be existing.
> 
> Which design guidelines?

[https://docs.microsoft.com/de-de/windows/uwp/design/app-settings/guidelines-for-app-settings#recommended-page-content](https://docs.microsoft.com/de-de/windows/uwp/design/app-settings/guidelines-for-app-settings#recommended-page-content)

That the shield is against the guidlines, is my interpretation. And I have never seen it in the modern Win10 Settings app.

But the guidlines say:
- If a control is disable you have to provide a description why above the control.
- The settings should show in every situation all/the same controls.

riverar commented:
> > 
> > Which design guidelines?
> 
> https://docs.microsoft.com/de-de/windows/uwp/design/app-settings/guidelines-for-app-settings#recommended-page-content
> 
> That the shield is against the guidlines, is my interpretation.

I don't see anything in here about the shield or images?

I agree though, Settings doesn't use a shield icon anywhere that I remember. Windows Security has elevation-required links (e.g. Firewall & Network Protection > Advanced Settings link at bottom) so I think their interpretation is "if elevation is needed, windows will take care of that". If we follow that pattern, I'd drop all hint text and just let the <kbd>Restart as administrator</kbd> button handle this. Or we can just agree Windows Security isn't a great example 😂

Perhaps we should move this to a dedicated issue.

htcfreek commented on src/core/Microsoft.PowerToys.Settings.UI/Views/GeneralPage.xaml:
```diff
@@ -92,6 +95,39 @@
+                        Style="{StaticResource SettingsGroupTitleStyle}"/>
+            <StackPanel Orientation="Horizontal">
+                <TextBlock Text="Version: " x:Uid="General_Version"  />
+                <HyperlinkButton NavigateUri="https://github.com/microsoft/PowerToys/releases" Content="{x:Bind ViewModel.PowerToysVersion }" Margin="4,-6,0,0">
```
Please move the version number from content attribut in a textblock. Then the text won't shown underlined anymore.

htcfreek commented:
> > > 
> > > Which design guidelines?
> > 
> > https://docs.microsoft.com/de-de/windows/uwp/design/app-settings/guidelines-for-app-settings#recommended-page-content
> > 
> > That the shield is against the guidlines, is my interpretation.
> 
> I don't see anything in here about the shield or images?
> 
> I agree though, Settings doesn't use a shield icon anywhere that I remember. Windows Security has elevation-required links (e.g. Firewall & Network Protection > Advanced Settings link at bottom) so I think their interpretation is "if elevation is needed, windows will take care of that". If we follow that pattern, I'd drop all hint text and just let the <kbd>Restart as administrator</kbd> button handle this. Or we can just agree Windows Security isn't a great example 😂
> 
> Perhaps we should move this to a dedicated issue.

Yes I agree that we should open a separate issue. Who should do this?

niels9001 commented:
> > > > Which design guidelines?
> > > 
> > > 
> > > https://docs.microsoft.com/de-de/windows/uwp/design/app-settings/guidelines-for-app-settings#recommended-page-content
> > > That the shield is against the guidlines, is my interpretation.
> > 
> > 
> > I don't see anything in here about the shield or images?
> > I agree though, Settings doesn't use a shield icon anywhere that I remember. Windows Security has elevation-required links (e.g. Firewall & Network Protection > Advanced Settings link at bottom) so I think their interpretation is "if elevation is needed, windows will take care of that". If we follow that pattern, I'd drop all hint text and just let the Restart as administrator button handle this. Or we can just agree Windows Security isn't a great example 😂
> > Perhaps we should move this to a dedicated issue.
> 
> Yes I agree that we should open a separate issue. Who should do this?

I'll open one up. Let's pause the discussion here on this topic and pick it up there. This PR is getting a bit polluted with things that were not part of the original issue.

crutkas commented:
For everyone here, please create a new issue.  A PR needs to be about the agreed upon work.

htcfreek commented:
@niels9001 

What about changing the toggle switch title "Download updates automatically" to "Download updates automatically when running with administrativ permission and you don't use a metered internet connection"? Then the user knows everything what is important and we don't need an explanation text under the toggle switch?
